File: plsql-frontend/src/test/java/org/sonar/plugins/plsqlopen/api/expressions/TreatAsExpressionTest.java
/*
 * Sonar PL/SQL Plugin (Community)
 * Copyright (C) 2015-2018 Felipe Zorzo
 * mailto:felipebzorzo AT gmail DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
package org.sonar.plugins.plsqlopen.api.expressions;

import org.junit.Before;
import org.junit.Test;
import org.sonar.plugins.plsqlopen.api.PlSqlGrammar;
import org.sonar.plugins.plsqlopen.api.RuleTest;

import static org.sonar.sslr.tests.Assertions.assertThat;

public class TreatAsExpressionTest extends RuleTest {

    @Before
    public void init() {
        setRootRule(PlSqlGrammar.EXPRESSION);
    }

    @Test
    public void matchesSimpleTreatAs() {
        assertThat(p).matches("treat(foo as bar)");
    }

    @Test
    public void matchesTreatAsRef() {
        assertThat(p).matches("treat(foo as ref bar)");
    }
    @Test
    public void matchesTreatAsRef2() {
        assertThat(p).matches("treat(foo as ref bar).test");
    }

}

File: zpa-core/src/test/kotlin/com/felipebz/zpa/api/expressions/TreatAsExpressionTest.kt
/**
 * Z PL/SQL Analyzer
 * Copyright (C) 2015-2025 Felipe Zorzo
 * mailto:felipe AT felipezorzo DOT com DOT br
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
package com.felipebz.zpa.api.expressions

import com.felipebz.flr.tests.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import com.felipebz.zpa.api.PlSqlGrammar
import com.felipebz.zpa.api.RuleTest

class TreatAsExpressionTest : RuleTest() {

    @BeforeEach
    fun init() {
        setRootRule(PlSqlGrammar.EXPRESSION)
    }

    @Test
    fun matchesSimpleTreatAs() {
        assertThat(p).matches("treat(foo as bar)")
    }

    @Test
    fun matchesTreatAsRef() {
        assertThat(p).matches("treat(foo as ref bar)")
    }

    @Test
    fun matchesTreatAsRef2() {
        assertThat(p).matches("treat(foo as ref bar).test")
    }

    @Test
    fun matchesAsExpression() {
        assertThat(p).matches("(foo as bar)")
    }

    @Test
    fun matchesAsExpressionWithMethodCall() {
        assertThat(p).matches("(foo as bar).baz()")
    }

}
